docs/proposals/hpa/federated-hpa-v2.md

```diff
@@ -0,0 +1,809 @@
+authors:
+- "@jwcesign" # Authors' GitHub accounts here.
+reviewers:
+- TBD
+approvers:
+- TBD
+
+creation-date: 2023-02-14
+
+---
+
+# Federated HPA
+## Summary
+HPA is a widely used approach to improve the stability of apps dealing with the burst of requests and resource utilization. We can run our apps on multiple clusters as easily as a single cluster with Karmada, but Karmada lacks HPA support now. 
+This proposal introduces a new component FederatedHPA to Karmada and enables Karmada to scale workloads across clusters with different policies. It also tries to minimize the user experience differences between HPA in a single cluster and multi-clusters.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.
+-->
+
+If users deploy the workloads in multi-clusters, they may want to scale the workloads in different clusters with different policies. For example, the workloads in the cluster with high priority may need to be scaled first before the clusters with low priority.
+
+### Goals
+1. Bring HPA from single cluster to multi-clusters.
+1. Compatible with the HPA-related resources in a single cluster.
+1. Tolerate the disaster of member clusters or Karmada control plane.
+1. It is better to integrate well with scenarios such as cloud burst.
+1. It is better to support both Kubernetes HPA and customized HPA.
+1. Minimize control plane overhead when scaling workloads across clusters, and maximize the utilization of member clusters' scaling capabilities. 
+1. Having rich scaling policies, users can enable different scenarios with different configurations.
+1. Consider cluster-autoscaler, scaling across clusters could be delayed after some period.
+
+### Non-Goals
+1. Compatible with the PP/CPP scheduling policy.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories(Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Migration HPA from single cluster to multi-clusters with minimal efforts
+As a platform developer using Kubernetes, now I want to use Karmada to run apps on multi clusters. But the CD ecosystem is built based on a single cluster and the original HPA is heavily used. So I want to migrate the HPA resources to multi-clusters without too much effort. It is better to be compatible with the schema of HPA used in a single cluster.
+
```

Comment:
Is the migration here automatic? 
Automatically convert user's HPA to Federated-HPA.

Reply:
No, you should do it manually.  But the spec is almost the same, so it will not cost much effort